quinn-proto/src/connection/mod.rs

```
@@ -362,6 +369,66 @@ impl Connection {
         this
     }
 
+    /// Set up qlog for this connection.
+    #[cfg(feature = "qlog")]
+    pub fn set_qlog(
+        &mut self,
+        writer: Box<dyn std::io::Write + Send + Sync>,
+        title: Option<String>,
+        description: Option<String>,
+        now: Instant,
+    ) {
+        let vp = if self.side.is_server() {
+            qlog::VantagePointType::Server
+        } else {
+            qlog::VantagePointType::Client
+        };
+
+        let level = qlog::events::EventImportance::Core;
+
+        let trace = qlog::TraceSeq::new(
+            qlog::VantagePoint {
+                name: None,
+                ty: vp,
+                flow: None,
+            },
+            title.clone(),
+            description.clone(),
+            Some(qlog::Configuration {
+                time_offset: Some(0.0),
+                original_uris: None,
+            }),
+            None,
+        );
+
+        let mut streamer = qlog::streamer::QlogStreamer::new(
+            qlog::QLOG_VERSION.to_string(),
+            title,
+            description,
+            None,
+            now,
+            trace,
+            level,
+            writer,
+        );
+
+        streamer.start_log().ok();
+
+        self.qlog_streamer = Some(streamer);
+    }
+
+    /// Emit a `MetricsUpdated` event to the qlog streamer containing only the updated values.
+    #[cfg(feature = "qlog")]
+    fn emit_qlog_recovery_metrics(&mut self, now: Instant) {
+        let Some(qlog_streamer) = &mut self.qlog_streamer else {
+            return;
+        };
+
+        let metrics = self.path.qlog_congestion_metrics(self.pto_count);
+        let event = qlog::events::EventData::MetricsUpdated(metrics);
+        let _ = qlog_streamer.add_event_data_with_instant(event, now);
```

What's the rationale for ignoring any errors from this?